### PR TITLE
vim-patch: update runtime files

### PR DESCRIPTION
**Pull request by clason**
- **vim-patch:af449f6: runtime(compiler): add cppcheck linter compiler plugin**
- **vim-patch:7c3f9af: runtime(misc): unset compiler in various ftplugins**

### CODE REVIEW

File: runtime/ftplugin/context.vim [hunk 1]
```diff
@@ -108,6 +104,11 @@ if get(g:, 'context_mappings', 1)
     endif
 endif
 
+if !exists('current_compiler')
+  b:undo_ftplugin ..= "| compiler make"
+  compiler context
+endif
```
Comment (zeertzjq) [inline runtime/ftplugin/context.vim:110]:
```suggestion
endif

```

File: runtime/ftplugin/context.vim [hunk 2]
```diff
@@ -108,6 +104,11 @@ if get(g:, 'context_mappings', 1)
...
+b:undo_ftplugin ..= "| sil! delc -buffer ConTeXt | sil! delc -buffer ConTeXtLog | sil! delc -buffer ConTeXtJobStatus | sil! delc -buffer ConTeXtStopJobs"
```
Comment (zeertzjq) [inline runtime/ftplugin/context.vim:111]:
```suggestion
  let b:undo_ftplugin ..= "| compiler make"
  compiler context
endif

let b:undo_ftplugin ..= "| sil! delc -buffer ConTeXt | sil! delc -buffer ConTeXtLog | sil! delc -buffer ConTeXtJobStatus | sil! delc -buffer ConTeXtStopJobs"
```

Comment (clason) [reply runtime/ftplugin/context.vim]: Sorry, forgot that this was vim9script. Should I drop the change entirely? I think that would be more consistent.

Comment (zeertzjq) [reply runtime/ftplugin/context.vim]: If you drop the change than it's inconsistent with other runtime files :)

**Review by zeertzjq — APPROVED** [review]

**Pull request merged**